fluster/utils.py

```diff
@@ -161,6 +161,32 @@ def normalize_path(path: str) -> str:
     return path
 
 
+def compare_yuv_files(reference_yuv: str, test_yuv: str, tolerance: int = 2, keep_files: bool = False) -> bool:
+    """Compare two YUV files pixel by pixel with the assertion abs(pixel_a - pixel_b) <= tolerance"""
+    with open(reference_yuv, "rb") as ref_file, open(test_yuv, "rb") as test_file:
+        ref_data = ref_file.read()
+        test_data = test_file.read()
+
+    if len(ref_data) != len(test_data):
+        raise ValueError(f"File sizes don't match - {len(ref_data)} vs {len(test_data)} bytes")
+
+    # Compare byte by byte (pixel by pixel)
+    total_violations = 0
+
+    for i, (pixel_ref, pixel_test) in enumerate(zip(ref_data, test_data)):
+        diff = abs(pixel_ref - pixel_test)
+        if diff > tolerance:
+            total_violations += 1
+
+    if not keep_files and os.path.isfile(test_yuv):
+        os.remove(test_yuv)
+
+    if total_violations > 0:
+        raise ValueError(f"Found {total_violations} pixels that don't meet tolerance requirement")
```

Comment:
The method returns a `bool`. I would expect that it returns `false` if the comparison fails. It's ok to raise an exception when the 2 files have a different size for example, but that would be an error somewhere else.

If you want to pass the number or errors to the caller, return an `int` with `total_violations`.